node-notes-cli/app.js

```diff
@@ -0,0 +1,86 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+const userInput = process.argv.slice(2);
+
+switch (userInput[0]) {
+  case 'read':
+    readNoteFunc();
+    break;
+  case 'create':
+    if (!userInput[1]) {
+      console.log("Empty input. Please enter a note.");
+    } else {
+      createNoteFunc();
+    }
+    break;
+  case 'delete':
+    if (!userInput[1]) {
+      console.log('Please enter a valid ID');
+    } else {
+      deleteNoteFunc();
+    }
+    break;
+  case 'update':
+    if (!userInput[2]) {
+      console.log('Please enter a new note to update.')
+    } else {
+      updateNoteFunc();
+    }
+    break;
+  default:
+    console.log('Input not recognized. Please provide valid input.');
+}
+
+// Display all notes in the data.json file
+async function readNoteFunc() {
+  try {
+    const content = JSON.parse(await readFile('data.json', 'utf8'));
```

Comment:
This code is repeated several times. It would be easier to maintain if this were extracted to a separate file.